views/application.py

```diff
@@ -93,3 +94,50 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+    View details of Application which should not be exposed directly
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self, request, *args, **kwargs):
+        """
+        Authenticate if the requesting user is Team Member
+        of the application and check the password entered.
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return: response containing client_secret
+        """
+
+        try:
+            app = Application.objects.get(pk=request.data.get('id'))
+
+            if app.team_members.filter(id=request.person.id).exists():
+
+                if request.user.check_password(request.data.get('password')):
+                    return response.Response(
+                        data=ApplicationHiddenDetailSerializer(app).data,
+                        status=status.HTTP_200_OK,
+                    )
+
+                else:
+                    return response.Response(
+                        data='Wrong Password',
```

Comment:
Be descriptive about this message. Better to use complete sentence instead of phrases. Keep special attention on the capitalisation.